src/Shared/Microsoft.AspNetCore.Razor.Utilities.Shared/PooledObjects/PooledArray`1.cs

```diff
@@ -9,17 +9,21 @@ namespace Microsoft.AspNetCore.Razor.PooledObjects;
 internal struct PooledArray<T> : IDisposable
 {
     private readonly ArrayPool<T> _pool;
+    private readonly int _minimumLength;
     private T[]? _array;
 
     // Because of how this API is intended to be used, we don't want the consumption code to have
     // to deal with Array being a nullable reference type. Instead, the guarantee is that this is
     // non-null until this is disposed.
     public readonly T[] Array => _array!;
 
+    public readonly Span<T> Span => _array!.AsSpan(0, _minimumLength);
```

Comment:
worth a comment that the span is the `_minimumLength` and not the full length of the `_array` for consumers to easily tell.